h5py/_hl/attrs.py

```diff
@@ -126,6 +126,9 @@ def create(self, name, data, shape=None, dtype=None):
             # First, make sure we have a NumPy array.  We leave the data type
             # conversion for HDF5 to perform.
             if not isinstance(data, Empty):
+                if numpy.size(data) > 3116:
+                    plist = self._id.get_create_plist()
+                    plist.set_attr_phase_change(0, 0)
```

Comment:
From the docs for [H5Pset_attr_phase_change](https://portal.hdfgroup.org/display/HDF5/H5P_SET_ATTR_PHASE_CHANGE), this is something that has to be done when creating the object/group to which the attributes belong, not when adding an attribute. `get_create_plist()` returns a *copy* of the creation property list - it can't be used to modify the object.

I'd recommend exposing the get/set functions in the low-level API first, without trying to use them in the high-level API.

Reply:
Ok, got it. thanks a lot~

Reply:
the problem is we create the group/dataset before creating attributes, and in many cases, we don't even create attributes, we can't know the value info about the attributes before we create it, so we can't decide how to set its value.